cantools/database/can/conversion.py

```diff
@@ -0,0 +1,190 @@
+        :return:
+            The calculated scaled value
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def scaled_to_raw(self, scaled_value: SignalValueType) -> Union[int, float]:
+        """Convert a scaled value to the internal raw value.
+
+        :param scaled_value:
+            The scaled value.
+        :return:
+            The internal raw value.
+        """
+        raise NotImplementedError
+
+
+class IdentityConversion(BaseConversion):
+    scale = 1
+    offset = 0
+    choices = None
+
+    def __init__(self, is_float: bool) -> None:
+        self.is_float = is_float
+
+    def raw_to_scaled(
+        self,
+        raw_value: Union[int, float],
+        decode_choices: bool = True,
+    ) -> Union[int, float]:
+        return raw_value
+
+    def scaled_to_raw(self, scaled_value: SignalValueType) -> Union[int, float]:
+        if not isinstance(scaled_value, (int, float)):
+            raise TypeError(
+                f"'scaled_value' must have type 'int' or 'float' (is {type(scaled_value)})"
+            )
+        return scaled_value
+
+
+class LinearIntegerConversion(BaseConversion):
+    is_float = False
+    choices = None
+
+    def __init__(self, scale: int, offset: int) -> None:
+        self.scale: int = scale
+        self.offset: int = offset
+
+    def raw_to_scaled(
+        self,
+        raw_value: Union[int, float],
+        decode_choices: bool = True,
+    ) -> SignalValueType:
+        return raw_value * self.scale + self.offset
+
+    def scaled_to_raw(self, scaled_value: SignalValueType) -> Union[int, float]:
+        if not isinstance(scaled_value, (int, float)):
+            raise TypeError(
+                f"'scaled_value' must have type 'int' or 'float' (is {type(scaled_value)})"
+            )
+        return round((scaled_value - self.offset) / self.scale)
+
+
+class LinearConversion(BaseConversion):
+    choices = None
+
+    def __init__(self, scale: float, offset: float, is_float: bool) -> None:
+        self.scale = scale
+        self.offset = offset
+        self.is_float = is_float
+
+    def raw_to_scaled(
+        self,
+        raw_value: Union[int, float],
+        decode_choices: bool = True,
+    ) -> SignalValueType:
+        return raw_value * self.scale + self.offset
+
+    def scaled_to_raw(self, scaled_value: SignalValueType) -> Union[int, float]:
+        if not isinstance(scaled_value, (int, float)):
+            raise TypeError(
+                f"'scaled_value' must have type 'int' or 'float' (is {type(scaled_value)})"
+            )
+        _raw = (scaled_value - self.offset) / self.scale
+        if self.is_float:
+            return _raw
+        return round(_raw)
+
+
+class NamedSignalConversion(BaseConversion):
+    is_float = False
+
+    def __init__(self, scale: float, offset: float, choices: Choices) -> None:
+        self.scale = scale
+        self.offset = offset
+        self._inverse_choices: Optional[Dict[str, int]] = None
```

Comment:
if the inverse choices are optional here, the choices should be too. alternatively, just declare the type of `self._inverse_choices` here:
```suggestion
        self._inverse_choices: Optional[Dict[str, int]]
```

Reply:
Fixed: Choices should be mandatory for NamedSignalConversion